addons/dragging/functions/fnc_startCarry.sqf

```diff
@@ -58,29 +61,32 @@ if (_target isKindOf "CAManBase") then {
     _unit action ["SwitchWeapon", _unit, _unit, 299];
     [_unit, "AmovPercMstpSnonWnonDnon", 0] call EFUNC(common,doAnimation);
 
-    // objects other than containers have calculated weight == 0 so we use getMass
-    if (-1 == ["ReammoBox_F", "WeaponHolder", "WeaponHolderSimulated"] findIf {_target isKindOf _x}) then {
-        _weight = getMass _target;
-    };
-    if (_weight > GVAR(maxWeightCarryRun)) then {
-        [_unit, "forceWalk", "ACE_dragging", true] call EFUNC(common,statusEffect_set);
-    } else {
-        [_unit, "blockSprint", "ACE_dragging", true] call EFUNC(common,statusEffect_set);
-    };
+    private _canRun = [_weight] call FUNC(canRun_carry);
+    // only force walking if we're overweight
+    [_unit, "forceWalk", "ace_dragging", !_canRun] call EFUNC(common,statusEffect_set);
+    [_unit, "blockSprint", "ace_dragging", _canRun] call EFUNC(common,statusEffect_set);
+
 };
 
-[_unit, "blockThrow", "ACE_dragging", true] call EFUNC(common,statusEffect_set);
+[_unit, "blockThrow", "ace_dragging", true] call EFUNC(common,statusEffect_set);
 
 // prevent multiple players from accessing the same object
 [_unit, _target, true] call EFUNC(common,claim);
 
-
 // prevents draging and carrying at the same time
 _unit setVariable [QGVAR(isCarrying), true, true];
 
 // required for aborting animation
 _unit setVariable [QGVAR(carriedObject), _target, true];
 
+// Event Handler for changes to inventory: allows unit to start running if weight is removed or be forced to walk if weight is added, drop object if weight past ACE_maxWeightCarry
+private _ehID = _target addEventHandler ["ContainerClosed", {
```

Comment:
event only fires on player accessing inventory
but we only add EH on the carrier's machine
I really don't think it's a problem, just noting that the CBA_fnc_targetEvent probably won't ever fire

adding and removing the EH too all machines in mp would be a big headache
could add EH globably via xeh and check if being dragged and has owner
but I dont think it's a big deal if weight changes mid-carry

Reply:
We can switch to `Put` and `Take` right now, it'll only have actual effect in 2.14, but should work in MP.
I misread the locality on `ContainerClosed` originally.

Reply:
Moved to XEH.